pkg/cmd/provisioning/aws/create-iam-roles_test.go

```diff
@@ -118,6 +158,40 @@ func TestIAMRoles(t *testing.T) {
 				assert.Equal(t, 1, provisioning.CountNonDirectoryFiles(files), "Should be exactly 1 secret in manifestsDir for one CredReq")
 			},
 		},
+		{
+			name:         "Ignore CredReq marked for deletion",
```

Comment:
I have included UT for only aws, but it covers the code path so we should be good.